File: benchmarks/NSubstitute.Analyzers.Benchmarks.CSharp.Source/DiagnosticsSources/NonSubstitutableMemberDiagnosticSource.cs
using NSubstitute.Analyzers.Benchmarks.Shared.Models;

namespace NSubstitute.Analyzers.Benchmarks.CSharp.Source.DiagnosticsSources
{
    public class NonSubstitutableMemberDiagnosticSource
    {
        public void NS1000_NonVirtualSetupSpecification()
        {
            var substitute = Substitute.For<Foo>();

            substitute.ObjectReturningMethod().Returns((IFoo)null);
            substitute.Property.Returns(1);
            substitute[0].Returns(1);
            1.Returns(1);

            SubstituteExtensions.Returns(substitute.ObjectReturningMethod(), (IFoo)null);
            SubstituteExtensions.Returns(substitute.Property, 1);
            SubstituteExtensions.Returns(substitute[0], 1);
            SubstituteExtensions.Returns(1, 1);
        }

        public void NS1003_InternalVirtualSetupSpecification()
        {
            var substitute = Substitute.For<Foo>();

            substitute.InternalObjectReturningMethod().Returns((IFoo)null);
            substitute.InternalObjectReturningProperty.Returns((IFoo)null);
            substitute[0, 0, 0].Returns((IFoo)null);

            SubstituteExtensions.Returns(substitute.InternalObjectReturningMethod(), (IFoo)null);
            SubstituteExtensions.Returns(substitute.InternalObjectReturningProperty, (IFoo)null);
            SubstituteExtensions.Returns(substitute[0, 0, 0], (IFoo)null);
        }
    }
}

File: benchmarks/NSubstitute.Analyzers.Benchmarks.VisualBasic.Source/DiagnosticsSources/NonSubstitutableMemberDiagnosticSource.vb
Imports NSubstitute.Analyzers.Benchmarks.VisualBasic.Source.Models

Namespace DiagnosticsSources
    Public Class NonSubstitutableMemberDiagnosticSource
        Public Sub NS1000_NonVirtualSetupSpecification()
            Dim substitute = NSubstitute.Substitute.[For](Of Foo)()
            substitute.ObjectReturningMethod().Returns(CType(Nothing, IFoo))
            substitute.[Property].Returns(1)
            substitute(0).Returns(1)
            SubstituteExtensions.Returns(substitute.ObjectReturningMethod(), CType(Nothing, IFoo))
            SubstituteExtensions.Returns(substitute.[Property], 1)
            SubstituteExtensions.Returns(substitute(0), 1)
            SubstituteExtensions.Returns(1, 1)
        End Sub

        Public Sub NS1003_InternalVirtualSetupSpecification()
            Dim substitute = NSubstitute.Substitute.[For](Of Foo)()
            substitute.InternalObjectReturningMethod().Returns(CType(Nothing, IFoo))
            substitute.InternalObjectReturningProperty.Returns(CType(Nothing, IFoo))
            substitute(0, 0, 0).Returns(CType(Nothing, IFoo))
            SubstituteExtensions.Returns(substitute.InternalObjectReturningMethod(), CType(Nothing, IFoo))
            SubstituteExtensions.Returns(substitute.InternalObjectReturningProperty, CType(Nothing, IFoo))
            SubstituteExtensions.Returns(substitute(0, 0, 0), CType(Nothing, IFoo))
        End Sub
    End Class
End Namespace
